backend/src/main/java/org/sefglobal/core/academix/service/CategoryService.java

```diff
@@ -27,4 +31,19 @@ public CategoryService(CategoryRepository categoryRepository) {
     public List<CustomCategory> getAllCategories(){
         return categoryRepository.findAllBy();
     }
+
+    /**
+     * Retrieves the category with the requested id
+     * @param categoryId which is the id of the requested category
+     * @return Category object
+     * @throws ResourceNotFoundException if a category with the requested id doesn't exist
+     */
+    public Category getCategoryById(Long categoryId) throws ResourceNotFoundException {
```

Comment:
Return a `CustomCategory` object.

Reply:
Changed the return type to `CustomCategory`